sysid-application/src/main/native/cpp/view/Generator.cpp

```diff
@@ -459,17 +460,23 @@ void Generator::Display() {
     }
   }
 
-  // Venom or NEO built-in encoders can't change sampling or measurement period.
+  // Venom built-in encoder, TalonFX Pro built-in encoder, and CANcoder Pro
+  // can't change sampling or measurement period.
   if (!((mainMotorController == sysid::motorcontroller::kVenom &&
          m_settings.encoderType == sysid::encoder::kBuiltInSetting) ||
-        (mainMotorController == sysid::motorcontroller::kSPARKMAXBrushless &&
-         m_settings.encoderType == sysid::encoder::kSMaxEncoderPort) ||
         (mainMotorController == sysid::motorcontroller::kTalonFXPro &&
          m_settings.encoderType == sysid::encoder::kBuiltInSetting) ||
         m_settings.encoderType == sysid::encoder::kCANcoderPro)) {
     // Samples Per Average Setting
     ImGui::SetNextItemWidth(ImGui::GetFontSize() * 2);
-    ImGui::InputInt("Samples Per Average", &m_settings.numSamples, 0, 0);
+    if (ImGui::InputInt("Samples Per Average", &m_settings.numSamples, 0, 0)) {
+      // Spark Max built-in encoder number of samples must be within 8 to 64
+      // inclusive
+      if (mainMotorController == sysid::motorcontroller::kSPARKMAXBrushless &&
+          m_settings.encoderType == sysid::encoder::kSMaxEncoderPort) {
+        m_settings.numSamples = std::clamp(m_settings.numSamples, 8, 64);
```

Comment:
Do we want to clamp or reject out-of-bounds here?

Reply:
As a rule, we clamp invalid inputs. To restore the original setting, we'd have to store what the previous value was.